python/triton/knobs.py

```diff
@@ -81,10 +97,7 @@ def from_env(self, val: str) -> SetType:
 class env_str(env_base[str, str]):
 
     def set(self, value: Optional[str]) -> None:
-        if value is None:
-            os.unsetenv(self.key)
-        else:
-            os.putenv(self.key, value)
+        setenv(self.key, value)
```

Comment:
But if `propagate_env` is False this doesn't set the value at all?

Reply:
Yeah that's right, that was intentional. In the least this is needed for testing, but I figured some users might want to opt out of the environment updates altogether (and just use the config classes as essentially a dataclass). Do you think that's ok? Or should we try to get rid of that flag (meaning I'd have to rethink the tests)?

Reply:
Oh I see there is `__set__`, `set` and `setenv` which all do subtly different things. Could `__set__` just be:
```python
    def __set__(self, obj: object, value: Union[SetType, Env]) -> None:
        if isinstance(value, Env):
            obj.__dict__.pop(self.name, None)
        else:
            obj.__dict__[self.name] = value
            setenv(self.key, repr(value) if value is not None else value)
```

Reply:
Yeah, good idea I think so-- it might need to look slightly different because e.g. [env_class](https://github.com/triton-lang/triton/blob/main/python/triton/knobs.py#L123) would need a custom `repr`